src/dataset_miner/data_extractor.py

```
@@ -0,0 +1,217 @@
+        for table in doc.tables:
+            content_parts.append("=== Table ===")
+            for row in table.rows:
+                row_text = " | ".join(cell.text.strip() for cell in row.cells)
+                if row_text.strip():
+                    content_parts.append(row_text)
+
+        text = "\n".join(content_parts)
+        logger.info(f"Extracted {len(text)} characters from {file_path}")
+        return text
+    except Exception as e:
+        raise TextExtractionError(f"Failed to extract text from DOCX: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)